test/code/plugins/oms_common_test.rb

```diff
@@ -290,6 +290,34 @@ def test_parse_json_record_encoding
       assert_equal("{\"DataItems\":[{\"Message\":\"iPhone®\"}]}", parsed_record, "parse json record utf-8 encoding failed");
     end	
     
+    def test_safe_dump_simple_hash_array
+      $log = MockLog.new
+
+      records = [ { "ID" => 1, "Enabled" => true, "Nil" => nil, "Float" => 1.2, "Message" => "iPhone®" } ];
+      json = Common.safe_dump_simple_hash_array(records);
+      assert_equal("[{\"ID\":1,\"Enabled\":true,\"Nil\":null,\"Float\":1.2,\"Message\":\"iPhone®\"}]", json, "parse json record utf-8 encoding failed: #{json}");
+
+      assert_equal(0, $log.logs.length, "No exception should be logged")
```

Comment:
Instead check that $log.logs is equal to an empty array. If it is not, then the unit test framework will print the difference and we will see the log messages.